File: agora-demo-ios-oc/AgoraDemo/Login/AGDLoginViewController.m
//
//  AGDLoginViewController.m
//  AgoraDemo
//
//  Created by apple on 15/9/9.
//  Copyright (c) 2015年 Agora. All rights reserved.
//

#import "AGDLoginViewController.h"
#import "AGDChatViewController.h"

static NSString * const AGDSegueIdentifierChat = @"Chat";

@interface AGDLoginViewController ()

@property (weak, nonatomic) IBOutlet UITextField *keyTextField;
@property (weak, nonatomic) IBOutlet UITextField *roomNumberTextField;

@end

@implementation AGDLoginViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
#ifdef AgoraDemo_IOS
#ifdef DEBUG
    self.keyTextField.text = @"341f3d3c30e94e13bd04be6601dfdd3d"; // Please use your own key. Test key could be forbidden in the future.
    self.roomNumberTextField.text = @"2048";
#endif
#endif
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (void)viewWillAppear:(BOOL)animated
{
    [super viewWillAppear:animated];
    self.navigationController.navigationBarHidden = YES;
    
    NSUserDefaults *userDefaults = [NSUserDefaults standardUserDefaults];
    NSString *vendorKey = [userDefaults objectForKey:AGDKeyVendorKey];
    if (vendorKey) {
        self.keyTextField.text = vendorKey;
    }
}

- (void)viewWillDisappear:(BOOL)animated
{
    [super viewWillDisappear:animated];
    self.navigationController.navigationBarHidden = NO;
}

#pragma mark -

- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender
{
    if (![segue.identifier isEqualToString:AGDSegueIdentifierChat]) {return;}
    
    AGDChatViewController *chatViewController = segue.destinationViewController;
    chatViewController.dictionary = @{AGDKeyChannel: self.roomNumberTextField.text,
                                      AGDKeyVendorKey: self.keyTextField.text};
    
    UIButton *button = (UIButton *)sender;
    if (button.tag == 1) {
        chatViewController.chatType = AGDChatTypeVideo;
    } else {
        chatViewController.chatType = AGDChatTypeAudio;
    }
}

#pragma mark -

- (IBAction)didClickPushChatViewController:(id)sender
{
    
    if ([self isValidInput]) {
        [self performSegueWithIdentifier:AGDSegueIdentifierChat sender:sender];
    }
}

- (BOOL)isValidInput
{
    [self.view endEditing:YES];
    if (_keyTextField.text.length && _roomNumberTextField.text.length) {
        return YES;
    } else {
        UIAlertView *alertView = [[UIAlertView alloc] initWithTitle:nil
                                                            message:NSLocalizedString(@"enter_key_room", nil)
                                                           delegate:nil
                                                  cancelButtonTitle:NSLocalizedString(@"done", nil)
                                                  otherButtonTitles:nil];
        [alertView show];
        return NO;
    }
}

- (IBAction)didTapHideKeyboard:(id)sender
{
    [self.view endEditing:YES];
}

@end

File: agora-demo-ios-swift/AgoraDemo/Login/AGDLoginViewController.swift
//
//  AGDLoginViewController.swift
//  AgoraDemo
//
//  Created by apple on 15/9/23.
//
//

import UIKit

class AGDLoginViewController: UIViewController {
    
    @IBOutlet weak var keyTextField: UITextField!
    @IBOutlet weak var roomNumberTextField: UITextField!
    
    override func viewDidLoad() {
        super.viewDidLoad()
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    override func viewWillAppear(animated: Bool) {
        super.viewWillAppear(animated)
        navigationController!.navigationBarHidden = true
        
        let userDefaults = NSUserDefaults.standardUserDefaults()
        let vendorKey = userDefaults.objectForKey(AGDKeyVendorKey)
        if vendorKey != nil {
            keyTextField.text = vendorKey as? String
        } else {
            // remove these code, and it's just for inner test
            let innerKeyUrl = NSURL(string: "http://192.168.99.253:8970/agora.inner.test.key.txt")
            if let jsonData: NSData = NSData(contentsOfURL: innerKeyUrl!) {
                let innerVendorKey = NSString(data: jsonData, encoding: NSUTF8StringEncoding)
                self.keyTextField.text = innerVendorKey!.stringByReplacingOccurrencesOfString("\n", withString: "") // Please use your own key. The inner test key is just invalid in public.
            }
        }
    }
    
    override func viewWillDisappear(animated: Bool) {
        super.viewWillDisappear(animated)
        navigationController!.navigationBarHidden = false
    }
    
    @IBAction func didTapHideKeyboard(sender: UITapGestureRecognizer) {
        view.endEditing(true)
    }
    
    @IBAction func didClickPushChatViewController(sender: UIButton) {
        
        view.endEditing(true)
        if isValidInput() {
            performSegueWithIdentifier(AGDKeySegueIdentifierChat, sender: sender)
        }
    }
    
    func isValidInput() -> Bool {
        
        if keyTextField.text!.isEmpty || roomNumberTextField.text!.isEmpty {
            var alertView = UIAlertView(title: nil, message:NSLocalizedString("enter_key_room",comment:"Please Enter Vendor Key and Room No.") , delegate: nil, cancelButtonTitle: NSLocalizedString("done",comment:"Done"))
            alertView.show()
            
            return false
        } else {
            return true
        }
    }
   
    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
        
        if segue.identifier == AGDKeySegueIdentifierChat {
            var chatViewController = segue.destinationViewController as! AGDChatViewController
            chatViewController.dictionary = [AGDKeyVendorKey:keyTextField.text!,AGDKeyChannel:roomNumberTextField.text!]
            
            var button = sender as! UIButton
            if button.tag == 1 {
                chatViewController.chatType = .Video
            } else {
                chatViewController.chatType = .Audio
            }
        }
    }
}
